### PR TITLE
[IDE] Fix a crash when closing a docknotebook tab

### PR DESCRIPTION
**Pull request by iainx**
The crash trace listed in the bug report doesn't say what the exception that caused the crash was, and I've
not been able to reproduce it, so I'll added a null check and a type check.

Fixes VSTS #706884

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs [hunk 1]
```diff
@@ -1354,7 +1354,9 @@ void CloseWindowEvent (object sender, WorkbenchWindowEventArgs e)
 		
 		internal void CloseClicked (object o, TabEventArgs e)
 		{
-			((SdiWorkspaceWindow)e.Tab.Content).CloseWindow (false, true).Ignore();
+			if (e.Tab.Content != null && e.Tab.Content is SdiWorkspaceWindow sdiWorkspace) {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs]: Can we log these two cases? Might be useful to know what went wrong.

**Comment by Therzok** [comment]
Is the exception trace in the bug report all of it? I'm wondering if that means closeclicked ran on a bg thread.

**Comment by iainx** [comment]
Yes, that's all of the CrashReport.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs (hunk 1 above)
Comment (iainx) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/DefaultWorkbench.cs]: Good idea, I noticed that while we treat `Tab.Content` as an `SdiWorkspaceWindow` everywhere, it's only enforced to be a `Gtk.Widget`.

**Comment by slluis** [comment]
@monojenkins backport release-7.7

**Pull request merged**